test/functorch/test_control_flow.py

```diff
@@ -1174,6 +1203,225 @@ def fwbw(map_op, f, x, y):
+
+        def mul(x: torch.Tensor, y: torch.Tensor):
+            return x * y
+
+        x = torch.randn(3, 10, 2, device=torch.device("cuda"))
+        for reverse in [False, True]:
+            for op, op_pt in [(add, torch.cumsum), (mul, torch.cumprod)]:
+                cumsum1 = associative_scan(op, x, 0, reverse=reverse)
+                cumsum_exp = _fake_associative_scan(op, x, 0, reverse=reverse)
+                self.assertEqual(cumsum1, cumsum_exp)
+                if not reverse:
+                    cumsum_exp_PT = op_pt(x, 0)
+                    self.assertEqual(cumsum1, cumsum_exp_PT)
+                else:
+                    # Check whether the graph contains at least two flip operations
+                    def f(op, inp, dim):
+                        result = associative_scan(op, inp, dim)
+                        return result
+
+                    # gm = make_fx(f, tracing_mode="symbolic")(op, x, 0)
+                    # self.assertExpectedInline(
+                    #     gm.code.strip(),
+                    #     """""",  # noqa: B950
+                    # )
+                    # self.assertNotRegex(
+                    #     gm.code.strip(),
+                    #     ".*.*",
+                    # )
+
+        # Jax Examples
+        x = torch.arange(0, 4, device=torch.device("cuda"))
+        cumsum1 = associative_scan(add, x, 0, reverse=False)
+        cumsum_exp = _fake_associative_scan(add, x, 0)
+        self.assertEqual(cumsum1, torch.tensor([0.0, 1.0, 3.0, 6.0], dtype=torch.int64))
+        self.assertEqual(cumsum1, cumsum_exp)
+
+        cumsum1 = associative_scan(add, x, 0, reverse=True)
+        cumsum_exp = _fake_associative_scan(add, x, 0, reverse=True)
+        self.assertEqual(cumsum1, torch.tensor([6.0, 6.0, 5.0, 3.0], dtype=torch.int64))
+        self.assertEqual(cumsum1, cumsum_exp)
+
+        num_dims = [random.randint(2, 5) for _ in range(10)]
+        for num_dim in num_dims:
+            shapes = [random.randint(1, 10) for _ in range(num_dim)]
+            rnd_scan_dim = random.randint(0, num_dim - 1)
+            x = torch.randn(*shapes, device=torch.device("cuda"))
+
+            for op, op_pt in [(add, torch.cumsum), (mul, torch.cumprod)]:
+                cumsum1 = associative_scan(op, x, rnd_scan_dim)
+                cumsum_exp = _fake_associative_scan(op, x, rnd_scan_dim)
+                self.assertEqual(cumsum1, cumsum_exp)
+                cumsum_exp_PT = op_pt(x, rnd_scan_dim)
+                self.assertEqual(cumsum1, cumsum_exp_PT)
+
+    @unittest.skipIf(not torch.cuda.is_available(), "Test requires CUDA.")
+    def test_generic_associative_scan_compile(self):
+        def add(x: torch.Tensor, y: torch.Tensor):
+            return x + y
+
+        def mul(x: torch.Tensor, y: torch.Tensor):
+            return x * y
+
+        x = torch.randn(3, 10, 2, device=torch.device("cuda"))
+        torch.compiler.reset()
+        with torch._dynamo.utils.disable_cache_limit():
```

Comment:
why is this necessary? I thought we already disabled cache limit in https://github.com/pytorch/pytorch/pull/133011/files#diff-43442eb487890d221156357e289c9ba39d7ee3da0a96bd7b13448b4716a4cfcaL78

Reply:
I removed the `with torch._dynamo.utils.disable_cache_limit():` statements